lib/helmsman/cluster/cluster.ex

```diff
@@ -10,7 +10,13 @@ defmodule Helmsman.Cluster do
     Enum.map(cluster_configs, &from_config/1)
   end
 
+  def determine_advertised_brokers(configured_brokers) do
+    KafkaEx.create_worker(:worker_foo, [uris: configured_brokers, consumer_group: "meh"])
+    KafkaEx.metadata(worker_name: :worker_foo).brokers
+    |> Enum.map(fn(broker) -> {broker.host, broker.port} end)
```

Comment:
There's nothing really wrong with this.  I think we're going to end up with a Metadata wrapper anyways so we can defer making it pretty to later.

If you want my anal retentive answer on how I'd write this to be the most style-correct i might do something like

```
metadata = KafkaEx.metadata(worker_name: :worker_foo)
Enum.map(metadata.brokers, fn(broker) -> {broker.host, broker.port} end)
```

You _could_ make the broker destructurerthing totally anonymous... it would look something like `&({&1.host, &1.name})` but I don't think that adds much.

Reply:
Don't pull any punches on me, I'll learn better if you're blunt... Be as anal retentive as you want.